core/templates/services/internet-connectivity.service.ts

```diff
@@ -61,6 +61,8 @@ export class InternetConnectivityService {
               return (
                 this._serverConnectionBackendApiService
                   .fetchConnectionCheckResultAsync());
+            } else {
+              throw Error();
```

Comment:
Why is this added and why is the error message empty?

Reply:
Initially the code was
```
switchMap(() => {
            if (this._connectedToNetwork) {
              return (
                this._serverConnectionBackendApiService
                  .fetchConnectionCheckResultAsync());
            }
          }),
```
so, if `this._connectedToNetwork` was false, `undefined` was being passed to `switchMap( )` (due to the missing else statement), however as switchMap expects an ObservableInput<any>, this threw an error while running strict checks.

As passing undefined to switchMap would throw an error (which is handled by `retryWhen`), throwing an Error in the else statement would satisfy the typescript checks, and not break anything. 

I left the error message empty, as I believe the message itself is not being used anywhere.

Reply:
Should we maybe use `NEVER` (from rxjs) instead?

Reply:
something like this?

```
else{
  return NEVER;
}
```

PS: I dont really have much experience with rxjs

Reply:
@vojtechjelinek using `NEVER` is still not satisfying the strict typescript checks
![Screenshot (7)](https://user-images.githubusercontent.com/73544247/133923579-d1647f29-29f8-47dd-aaed-6bc33402ca16.png)

Reply:
Can you try returning it?

Reply:
Yes I did try it, it passes the typescript test, but not the frontend tests.
![Screenshot (8)](https://user-images.githubusercontent.com/73544247/133993838-74240ba8-34c9-41a5-9043-062ed2513436.png)

Reply:
Hmm, not sure about this @srijanreddy98 @mridul-netizen @EeshaArif would you be able to help here?

Reply:
@atpug22 can you take a look here? Should `NEVER` be returned here. Please take a look at test failure as well.Thanks!

Reply:
I think a better option should be to use `EMPTY` but it will result in the same test failure. We might need to update the test itself (not sure about this).

Reply:
Can you test the behaviour locally and upload a video of that? It should be enough to disable the connection in DevTools